software/bootcamp-code/motion.py

```diff
@@ -0,0 +1,156 @@
+import struct
+import turtle
+import math
+import numpy as np
+import time
+import tkinter as tk
+import serial
+import serial.tools.list_ports
+import math
+
+class IRobotMotion:
+    def open(self):
+        pass
+    def close(self):
+        pass
+    def move(self, x_speed, y_speed, rot_speed):
+        pass
+
+# This code is added by Pille P2rnalaas
+# class OmniMotionRobot implements the IRobotMotion interface
+class OmniMotionRobot(IRobotMotion): #extension of the IRobotMotion
+    def __init__(self, name="Omni motion robot"):
+        self.ser = None
+        # Wheel angles
+        self.motor_config = [0, 120, 240]
+        print("initializing"+name)
+
+    def open(self):
+        print("Open OmniMotionRobot")
+        port=None
+        # get all the available seiral ports
+        ports = serial.tools.list_ports.comports()
+        # find the correct serial port
+        for p in ports:
+            print(p.device)
+            print(p.hwid)
+            # idProduct=5740 to filter the list given by serial.tools.list_ports.comports(include_links=False)
+            if p.hwid.__contains__("5740"):
+                port = p.device
+                print("Serial port assigned")
+                break
+        # open the serial connection and initialize the serial session (self.ser)
+        self.ser = serial.Serial(port)
+    def close(self):
+        print("Close OmniMotionRobot")
+        self.ser.close()
+
+    # new method for sending commands to the mainboard over serial
+    # Speed values are between 48 - 2047 for 0 to 100% speed
+    def send_commands(self, speed1=0, speed2=0, speed3=0, thrower_speed=0):
+        disable_failsafe = 0
+        command = struct.pack('<hhhHBH', speed1, speed2, speed3, thrower_speed, disable_failsafe, 0xAAAA)
+        self.ser.write(command)
+
+    def move(self, x_speed, y_speed, rot_speed):
+        speeds = [0, 0, 0]
+        # wheels distance from the center in m
+        # wheelDistance = 0.125
+
+        # TODO This is where you need to calculate the speeds for robot motors
+        robotSpeed = math.sqrt(x_speed*x_speed + y_speed*y_speed)
+        robotDirectionAngle = math.atan2(y_speed, x_speed)
+        # robot angular speed is basically robot rotational speed
+        robotAngularSpeed = rot_speed
+        for s in speeds:
```

Comment:
for s in speeds returns the elements of the list, no?
![image](https://user-images.githubusercontent.com/5435954/195283084-4112489b-036c-4a31-b7ac-0815aa2b2d02.png)
You most seem to use want to use the list indexes. In this case you can use the `for i in range(3)` 
![image](https://user-images.githubusercontent.com/5435954/195283446-93b9c492-b170-4b3b-8cf8-27da6b0cc5d5.png)